projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -443,34 +443,94 @@ function wpcom_launchpad_track_completed_task( $task_id, $extra_props = array()
 	);
 }
 
+/**
+ * Returns a list of mappings from 'id_map' to the actual task id.
+ *
+ * @return array
+ */
+function wpcom_launchpad_get_reverse_id_mappings() {
+	$task_definitions = wpcom_launchpad_get_task_definitions();
+
+	$mapping = array();
+	foreach ( $task_definitions as $task_id => $value ) {
+		if ( ! isset( $value['id_map'] ) ) {
+			continue;
+		}
+		$mapping[ $value['id_map'] ] = $task_id;
+	}
+
+	return $mapping;
+}
 /**
  * Mark a task as complete.
  *
- * @param string $task_id The task ID.
+ * @param string|string[] $task_ids The task IDs that should be marked as complete.
  * @return bool True if the task was marked as complete, false otherwise.
  */
-function wpcom_mark_launchpad_task_complete( $task_id ) {
+function wpcom_mark_launchpad_task_complete( $task_ids ) {
+	$task_ids = is_array( $task_ids ) ? $task_ids : array( $task_ids );
+	$values   = array_fill_keys( $task_ids, true );
+
+	$result = wpcom_launchpad_update_task_status( $values );
+
+	return $result;
+}
+
+/**
+ * Mark a task as incomplete. This method is here for backwards compatibility of
+ * the Rest API that allowed marking a task as incomplete.
+ *
+ * @param string|string[] $task_ids The task IDs that should be marked as complete.
+ * @return bool True if the task was marked as complete, false otherwise.
+ */
+function wpcom_mark_launchpad_task_incomplete( $task_ids ) {
+	$task_ids = is_array( $task_ids ) ? $task_ids : array( $task_ids );
+	$values   = array_fill_keys( $task_ids, false );
+
+	return wpcom_launchpad_update_task_status( $values );
+}
+
+/**
+ * Updates task/s statuses. If a non-existent task is passed, it will be ignored.
+ * It will trigger the completion event in Tracks for the tasks that were marked as complete.
+ *
+ * @param array $new_statuses Array of mappings [ task_id: string => status: bool ].
+ * @return bool True if the option where the statuses are stored was updated, false otherwise.
```

Comment:
Might it make sense for this function to return an array of boolean values with the task IDs as the keys? It's going to be hard to clearly map a set of actions back to a single boolean. (This is not blocking, BTW.)

Reply:
I kept this because this was what the endpoint was returning. If we are changing that this is a good change! Probably makes sense to make it in a separate PR so it's easier to test.

Reply:
I modified the API to handle only one incoming task, so this question is moot.